src/Validation/Validator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation;
+
+use Lcobucci\JWT\Token;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ *
+ * @since 4.0.0
+ */
+final class Validator implements \Lcobucci\JWT\Validator
```

Comment:
```php
$validator = new Validator();
$validator->validate($token, ...$constraints)->isValid()
```
**Eek, 3 times *valid* on a single line. Wait – 3 times?**

The `Validator` class is final and doesn't have any state, thus it is only a *container* or the *namespace* for the `validate` and `assert` functions, isn't it? For consumers, this is very inconvenient, I think they should use those functions directly:
```php
use function Lcobucci\JWT\assert;
use function Lcobucci\JWT\validate;

assert($token, ...$constraints);
$result = validate($token, ...$constraints); // whatever result will be
```

Btw, because of this, I don't know if the `Validator` class and the `Validator` interface are need at all. Interop? – Questionable. Of course, they can be kept for internal usage, but I think providing only functions is much more appropriate.